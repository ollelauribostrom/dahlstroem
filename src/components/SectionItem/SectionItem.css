.section-item {
  display: block;
}

.section-item--link {
  color: #454dc1;
}
