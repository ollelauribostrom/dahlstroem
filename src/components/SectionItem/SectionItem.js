import React from 'react';
import PropTypes from 'prop-types';

const SectionItem = ({ description, href }) => {
  if (href) {
    return (
      <a
        href={href}
        target="_blank"
        rel="noopener noreferrer"
        title={description}
        className="section-item section-item--link"
      >
        {description}
      </a>
    );
  }
  return <span className="section-item">{description}</span>;
};

SectionItem.propTypes = {
  description: PropTypes.string.isRequired,
  href: PropTypes.string,
};

export default SectionItem;
