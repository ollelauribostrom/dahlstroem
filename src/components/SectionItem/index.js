import SectionItem from './SectionItem';
import './SectionItem.css';

export default SectionItem;
