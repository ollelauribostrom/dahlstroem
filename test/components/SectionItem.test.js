import React from 'react';
import { shallow } from 'enzyme';

import SectionItem from '../../src/components/SectionItem';

describe('Tests for SectionItem', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SectionItem description="test" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly when href prop is provided', () => {
    const wrapper = shallow(
      <SectionItem description="test" href="google.com" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
