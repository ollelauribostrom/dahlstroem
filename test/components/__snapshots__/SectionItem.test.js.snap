// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for SectionItem renders correctly 1`] = `
<span
  className="section-item"
>
  test
</span>
`;

exports[`Tests for SectionItem renders correctly when href prop is provided 1`] = `
<a
  className="section-item section-item--link"
  href="google.com"
  rel="noopener noreferrer"
  target="_blank"
  title="test"
>
  test
</a>
`;
